docs/docs/loading-data-asynchronously.md

```diff
@@ -0,0 +1,155 @@
+
+Building on the previous example, we will pass the username as a prop instead and fetch new repos when it changes:
+
+```javascript{7-11,,13-15,17-21,35,38,44-47,53}
+class Repos extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {repos: []};
+  }
+
+  fetchRepos() {
+    axios.get(`https://api.github.com/users/${this.props.username}/repos`)
+      .then(response => this.setState({ repos: response.data }))
+      .catch(error => console.log(error));
+  }
+
+  componentDidMount() {
+    this.fetchRepos();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.username != prevProps.username) {
+      this.fetchRepos();
+    }
+  }
+
+  render() {
+    return (
+      <div>
+        <h1>Repos by {this.props.username}</h1>
+        {this.state.repos.map(repo =>
+          <div key={repo.id}>{repo.name}</div>
+        )}
+      </div>
+    );
+  }
+}
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {username: 'facebook'};
+  }
+
+  render() {
+    return (
+      <div>
+        <button onClick={() => this.setState({username: 'facebook'})}>Facebook</button>
+        <button onClick={() => this.setState({username: 'microsoft'})}>Microsoft</button>
+        <button onClick={() => this.setState({username: 'google'})}>Google</button>
+        <Repos username={this.state.username} />
+      </div>
+    );
+  }
+}
+
+ReactDOM.render(<App />, document.getElementById('app'))
+```
+
+[Try it on CodePen.](http://codepen.io/dashtinejad/pen/zNpzVW?editors=0011)
+
+## Cancellation
+
+An old promise can be pending when a newer promise fulfills. This can cause the old promise to override the result of the new one. If a promise is pending when a component is updated, the result of the first promise should be ignored before a new one is created.
+Some data fetching APIs allow you to cancel requests, and for axios, we use [Cancellation](https://github.com/mzabriskie/axios#cancellation) by token:
+
+```javascript{5-8,10-11,14-15,19-23}
+class Repos extends React.Component {
+  // removed for brevity
+
+  fetchRepos() {
+    // cancel the previous request
+    if (typeof this._source != typeof undefined) {
+      this._source.cancel('Operation canceled due to new request.')
+    }
+
+    // save the new request for cancellation
+    this._source = axios.CancelToken.source();
+    
+    axios.get(`https://api.github.com/users/${this.props.username}/repos`,
+      // cancel token used by axios
+      { cancelToken: this._source.token }
+    )
+      .then(response => this.setState({ repos: response.data }))
+      .catch(error => {
+         if (axios.isCancel(error)) {
+           console.log('Request canceled', error);
+         } else {
+           console.log(error);
```

Comment:
state error - show error message. Not that much code, but will encourage programmers to implement error messages and think of all states of component with async load